miden-lib/build.rs

```diff
@@ -205,6 +205,15 @@ fn get_masm_files<P: AsRef<Path>>(dir_path: P) -> io::Result<Vec<PathBuf>> {
 /// # Errors
 /// Returns an error if the path could not be converted to a UTF-8 string.
 fn is_masm_file(path: &Path) -> io::Result<bool> {
+    if path
+        .file_name()
+        .and_then(|name| name.to_str())
+        .map(|name| name.eq_ignore_ascii_case("api.masm"))
+        == Some(true)
+    {
+        return Ok(false);
+    }
```

Comment:
Instead of doing it here, I would create a dedicated function for compiling kernels and then call this function on line 51 above. The function would be pretty simple now - it would just compile `main.rs` for transaction kernel.

Reply:
It's cleaner, but it adds a bit more code.